src/tests/class_tests/openms/source/PeakIntegrator_test.cpp

```diff
@@ -848,6 +920,459 @@ START_SECTION(PeakShapeMetrics calculatePeakShapeMetrics(
+  );
+  TEST_REAL_SIMILAR(diff_E_param, 0.0)
+  TEST_REAL_SIMILAR(param_lr, 4.0)
+  TEST_REAL_SIMILAR(param_update, -4.0)
+  TEST_REAL_SIMILAR(param, 856.0)
+}
+END_SECTION
+
+START_SECTION(double compute_z(
+  const double x,
+  const double mu,
+  const double sigma,
+  const double tau
+) const)
+{
+  PeakIntegrator_friend pi_f;
+  double x;
+  const double mu { 14.3453 };
+  const double sigma { 0.0344277 };
+  const double tau { 0.188507 };
+
+  x = mu - 1.0 / 60.0;
+  TEST_REAL_SIMILAR(pi_f.compute_z(x, mu, sigma, tau), 0.471456263584609)
+
+  x = mu + 1.0 / 60.0;
+  TEST_REAL_SIMILAR(pi_f.compute_z(x, mu, sigma, tau), -0.213173439809831)
+
+  x = -3333333;
+  TEST_REAL_SIMILAR(pi_f.compute_z(x, mu, sigma, tau), 68463258.2588395)
+}
+END_SECTION
+
+START_SECTION(double emg_point(
+  const double x,
+  const double h,
+  const double mu,
+  const double sigma,
+  const double tau
+) const)
+{
+  PeakIntegrator_friend pi_f;
+  double x;
+  double h { 15515900 };
+  double mu { 14.3453 };
+  double sigma { 0.0344277 };
+  double tau { 0.188507 };
+
+  x = mu - 1.0 / 60.0;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 1992032.65711041)
+
+  x = mu + 1.0 / 60.0;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 4088964.97520213)
+
+  x = -3333333;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 0.0)
+
+  mu = 860.719;
+  sigma = 2.06566;
+  tau = 11.3104;
+
+  x = mu - 1;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 1992033.06584247)
+
+  x = mu + 1;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 4088968.52957875)
+
+  x = -200000000;
+  TEST_REAL_SIMILAR(pi_f.emg_point(x, h, mu, sigma, tau), 0.0)
+}
+END_SECTION
+
+START_SECTION(void emg_vector(
+  const std::vector<double>& xs,
+  const double h,
+  const double mu,
+  const double sigma,
+  const double tau,
+  std::vector<double>& out_xs,
+  std::vector<double>& out_ys
+) const)
+{
+  PeakIntegrator_friend pi_f;
+  const double h { 15515900 };
+  const double mu { 14.3453 };
+  const double sigma { 0.0344277 };
+  const double tau { 0.188507 };
+  vector<double> out_xs;
+  vector<double> out_ys;
+
+  Param params = pi_f.peakIntegrator.getParameters();
+
+  params.setValue("compute_additional_points", "false");
+  pi_f.peakIntegrator.setParameters(params);
+  pi_f.emg_vector(saturated_cutoff_pos_min, h, mu, sigma, tau, out_xs, out_ys);
+  TEST_EQUAL(out_xs.size(), saturated_cutoff_pos_min.size())
+  TEST_REAL_SIMILAR(out_xs.front(), 14.3310337)
+  TEST_REAL_SIMILAR(out_ys.front(), 2144281.1472228)
+
+  params.setValue("compute_additional_points", "true");
+  pi_f.peakIntegrator.setParameters(params);
+  pi_f.emg_vector(saturated_cutoff_pos_min, h, mu, sigma, tau, out_xs, out_ys);
+  TEST_EQUAL(out_xs.size(), 71) // more points than before
+  TEST_REAL_SIMILAR(out_xs.front(), 14.2717555076923) // peak was cutoff on the left side
+  TEST_REAL_SIMILAR(out_ys.front(), 108845.941990663)
+}
+END_SECTION
+
```

Comment:
Nice work adding test coverage for all of the private/helper functions.  That will help greatly when future researchers look at this code, and for when/if we need to make changes/modifications.